ruby/ql/lib/codeql/ruby/ApiGraphs.qll

```diff
@@ -803,5 +773,37 @@ module API {
 
     /** Gets the label for the edge from the root node to a custom entry point of the given name. */
     LabelEntryPoint entryPoint(API::EntryPoint name) { result.getName() = name }
+
+    /** Gets the API graph label corresponding to the given argument position. */
+    Label::ApiLabel getLabelFromArgumentPosition(DataFlowDispatch::ArgumentPosition pos) {
+      exists(int n |
+        pos.isPositional(n) and
+        result = Label::parameter(n)
+      )
+      or
+      exists(string name |
+        pos.isKeyword(name) and
+        result = Label::keywordParameter(name)
+      )
+      or
+      pos.isBlock() and
+      result = Label::blockParameter()
```

Comment:
Should the `toString` of `LabelBlockParameter` be updated to `getParameter(block)` (and similar for `LabelKeywordParameter`? And are we missing labels for the `self` parameter?

Reply:
The `toString` value of these labels are currently named after predicates on `API::Node` (not MaD tokens). At least that seemed to be the convention, one that I've tried to continue. So it doesn't need updating.

Reply:
I like having the `toString()` be the associated predicate on `API::Node`, so maybe we should update JS to also do that?